src/tools/wasm-link.cc

```diff
@@ -169,14 +169,20 @@ static uint32_t relocate_func_index(LinkerInputBinary* binary,
   } else {
     /* imported function call */
     FunctionImport* import = &binary->function_imports[function_index];
-    offset = binary->imported_function_index_offset;
+    uint32_t new_index = import->relocated_function_index;
     if (!import->active) {
       function_index = import->foreign_index;
       offset = import->foreign_binary->function_index_offset;
       if (s_debug)
         writef(&s_log_stream,
                "reloc for disabled import. new index = %d + %d\n",
                function_index, offset);
```

Comment:
I thought so at first, but actually we hit this code if we resolve a relocation internally. If a.wasm defines a.f and b.wasm imports a.f and calls it, the call relocation will end up in this code.